CMakeModules/CPackConfig.cmake

```diff
@@ -131,81 +131,55 @@ cpack_add_install_type(Extra DISPLAY_NAME "Extra")
 
-if (PACKAGE_MKL_DEPS AND PACKAGE_GFX_DEPS)
-  cpack_add_component(cpu
-    DISPLAY_NAME "CPU Backend"
-    DESCRIPTION "This Backend allows you to run ArrayFire code on native CPUs."
-    DEPENDS mkl_dependencies gfx_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(opencl
-    DISPLAY_NAME "OpenCL Backend"
-    DESCRIPTION "This Backend allows you to take advantage of OpenCL capable GPUs to run ArrayFire code. Currently ArrayFire does not support OpenCL for the Intel CPU on OSX."
-    DEPENDS mkl_dependencies gfx_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(cuda
-    DISPLAY_NAME "CUDA Backend"
-    DESCRIPTION "This Backend allows you to take advantage of the CUDA enabled GPUs to run ArrayFire code. Please make sure you have CUDA toolkit installed or install CUDA dependencies component."
-    DEPENDS gfx_dependencies cuda_dependencies
-    INSTALL_TYPES Development Runtime)
-elseif (PACKAGE_MKL_DEPS)
-  cpack_add_component(cpu
-    DISPLAY_NAME "CPU Backend"
-    DESCRIPTION "This Backend allows you to run ArrayFire code on native CPUs."
-    DEPENDS mkl_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(opencl
-    DISPLAY_NAME "OpenCL Backend"
-    DESCRIPTION "This Backend allows you to take advantage of OpenCL capable GPUs to run ArrayFire code. Currently ArrayFire does not support OpenCL for the Intel CPU on OSX."
-    DEPENDS mkl_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(cuda
-    DISPLAY_NAME "CUDA Backend"
-    DESCRIPTION "This Backend allows you to take advantage of the CUDA enabled GPUs to run ArrayFire code. Please make sure you have CUDA toolkit installed or install CUDA dependencies component."
-    DEPENDS cuda_dependencies
-    INSTALL_TYPES Development Runtime)
-elseif (PACKAGE_GFX_DEPS)
-  cpack_add_component(cpu
-    DISPLAY_NAME "CPU Backend"
-    DESCRIPTION "This Backend allows you to run ArrayFire code on native CPUs."
-    DEPENDS gfx_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(opencl
-    DISPLAY_NAME "OpenCL Backend"
-    DESCRIPTION "This Backend allows you to take advantage of OpenCL capable GPUs to run ArrayFire code. Currently ArrayFire does not support OpenCL for the Intel CPU on OSX."
-    DEPENDS gfx_dependencies
-    INSTALL_TYPES Development Runtime)
-  cpack_add_component(cuda
-    DISPLAY_NAME "CUDA Backend"
-    DESCRIPTION "This Backend allows you to take advantage of the CUDA enabled GPUs to run ArrayFire code. Please make sure you have CUDA toolkit installed or install CUDA dependencies component."
-    DEPENDS gfx_dependencies cuda_dependencies
-    INSTALL_TYPES Development Runtime)
+cpack_add_component(cuda
+  DISPLAY_NAME "CUDA Backend"
+  DESCRIPTION "This Backend allows you to take advantage of the CUDA enabled GPUs to run ArrayFire code. Please make sure you have CUDA toolkit installed or install CUDA dependencies component."
```

Comment:
I don't think the part about installing the cuda toolkit is relevant anymore

Reply:
If the user chose to not install `CUDA Dependencies` component in our installer, then they have to install the `cu*` libraries themselves. I thought that is entirely the reason you asked me to provide an option for `CUDA dependencies` component to be deselected if the user chooses to.